pymongo/asynchronous/mongo_client.py

```diff
@@ -1552,24 +1677,25 @@ async def close(self) -> None:
         .. versionchanged:: 3.6
            End all server sessions created by this client.
         """
-        session_ids = self._topology.pop_all_sessions()
-        if session_ids:
-            await self._end_sessions(session_ids)
-        # Stop the periodic task thread and then send pending killCursor
-        # requests before closing the topology.
-        self._kill_cursors_executor.close()
-        await self._process_kill_cursors()
-        await self._topology.close()
-        if self._encrypter:
-            # TODO: PYTHON-1921 Encrypted MongoClients cannot be re-opened.
-            await self._encrypter.close()
-        self._closed = True
-        if not _IS_SYNC:
-            await asyncio.gather(
-                self._topology.cleanup_monitors(),  # type: ignore[func-returns-value]
-                self._kill_cursors_executor.join(),  # type: ignore[func-returns-value]
-                return_exceptions=True,
-            )
+        if self._topology is not None:
```

Comment:
Rather than indenting this code (which causes code churn) how about we do this?:
```python
if self._topology is not None:
    return
session_ids = self._topology.pop_all_sessions()
...
```